apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts

```diff
@@ -0,0 +1,33 @@
+import { ILivechatVisitor } from '@rocket.chat/core-typings';
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+
+import { parseOutboundPhoneNumber } from '../../../ee/client/lib/voip/parseOutboundPhoneNumber';
+
+type Contact = {
+	name: string;
+	phone: string;
+};
+
+const createContact = (phone: string, data: Pick<ILivechatVisitor, 'name'> | null): Contact => ({
+	phone,
+	name: data?.name || '',
+});
+
+export const useOmnichannelContact = (ogPhone: string, name = ''): Contact => {
+	const getContactBy = useEndpoint('GET', '/v1/omnichannel/contact.search');
+	const phone = parseOutboundPhoneNumber(ogPhone);
+	const [defaultContact] = useState<Contact>({ phone, name });
+
+	const {
+		data: contact,
+		isLoading,
+		isError,
+	} = useQuery(['getContactsByPhone', phone], async (): Promise<Contact> => {
+		const { contact } = await getContactBy({ phone });
+		return createContact(phone, contact);
+	});
+
+	return isLoading || isError || !contact ? defaultContact : contact;
```

Comment:
I don't like to omit hook states (loading/error). Why not return the entire query? instead of the default value?

Reply:
@ggazzo The idea is that if you are using this hook you can trust it to always return a contact. You don't have to worry about what's happening behind the scenes (loading/error). I can return those states, no problem, but at least for now I don't see a scenario where I would need them. I believe a good compromise is returning and something like `{ isLoading: boolean; isError: boolean; contact: Contact }`, wdyt?